blaze-client/src/test/scala/org/http4s/blaze/client/BlazeClientSuite.scala

```diff
@@ -244,4 +246,25 @@ class BlazeClientSuite extends BlazeClientBase {
       }
       .assert
   }
+
+  test("Blaze HTTP/1 client should raise a ResponseException when it receives an unexpected EOF") {
+    SocketGroup[IO](testBlocker).use {
+      _.serverResource[IO](new InetSocketAddress(0))
+        .map { case (addr, sockets) =>
+          val uri = Uri.fromString(s"http://[${addr.getHostName}]:${addr.getPort}/eof").yolo
```

Comment:
Our URI is a damn disgrace.  This test will be flaky if an IPv6 address isn't returned.